source/Octopus.Tentacle.Tests/Kubernetes/ReferenceCountingKeyedBinarySemaphoreTests.cs

```diff
@@ -1,128 +1,66 @@
+    [Timeout(20000)] //Timeout test after 20 seconds
     public class ReferenceCountingKeyedBinarySemaphoreTests
     {
-        CancellationTokenSource testCancellationTokenSource;
-        ConcurrentDictionary<int, int> referenceCountIntercepts;
-
-        [SetUp]
-        public void SetUp()
-        {
-            testCancellationTokenSource = new CancellationTokenSource(TimeSpan.FromMinutes(10));
-            referenceCountIntercepts = new ConcurrentDictionary<int, int>();
-        }
-
-        [TearDown]
-        public void TearDown()
-        {
-            testCancellationTokenSource.Dispose();
-        }
-
-        [TestCase(1)]
-        [TestCase(2)]
-        [TestCase(20)]
-        public async Task MultipleThreadsLockWithSameKey_OnDisposeCalledOnceForEachThread(int numThreads)
-        {
-            // Arrange
-            var keyedLock = new TestableReferenceCountingKeyedBinarySemaphore<ScriptTicket>(RecordingReleaserFactory);
-
-            // Act
-            var tasks = Enumerable.Range(1, numThreads).Select(_ => Task.Run(async () => await SimulateWork(keyedLock, "foo", testCancellationTokenSource.Token)));
-            await tasks.WhenAll();
-
-            // Assert
-            var expected = Enumerable.Range(0, numThreads).ToDictionary(x => x, _ => 1);
-            referenceCountIntercepts.Should().BeEquivalentTo(expected);
-        }
-
         [Test]
-        public async Task ThreadsLockWithDifferentKeys_OnDisposeCalledWithCorrectReferenceCount()
+        public async Task WaitingOnTheSameKey_BlocksSuccessiveAttempts()
         {
             // Arrange
-            var keyedLock = new TestableReferenceCountingKeyedBinarySemaphore<ScriptTicket>(RecordingReleaserFactory);
-
-            // Act
-            var tasks = new[]
-            {
-                Task.Run(async () => await SimulateWork(keyedLock, "banana", testCancellationTokenSource.Token)),
-                Task.Run(async () => await SimulateWork(keyedLock, "pineapple", testCancellationTokenSource.Token)),
-                Task.Run(async () => await SimulateWork(keyedLock, "banana", testCancellationTokenSource.Token))
-            };
-            await tasks.WhenAll();
-
-            // Assert
-            var expected = new Dictionary<int, int>
-            {
-                { 0, 2 }, // banana, pineapple
-                { 1, 1 } // banana
-            };
-            referenceCountIntercepts.Should().BeEquivalentTo(expected);
-        }
-
-        async Task SimulateWork(IKeyedSemaphore<ScriptTicket> keyedSemaphore, string taskId, CancellationToken cancellationToken)
-        {
-            using (await keyedSemaphore.WaitAsync(new ScriptTicket(taskId), testCancellationTokenSource.Token))
+            var referenceCountingSemaphore = new ReferenceCountingKeyedBinarySemaphore<string>();
+            var sharedKey = "SomeSharedKey";
+            var periodToWaitForLock = TimeSpan.FromSeconds(2); // How long should the task wait to attempt to acquire the lock.
```

Comment:
Its entirely possible that this could timeout before we have guaranteed that it has attempted to get the lock, but without overcomplicating the tests it can sometimes be easier to test threading code by validating what it _cant_ do, rather than what it can.